### PR TITLE
fix(forge): record snapshot failures correctly

### LINKED ISSUE #3055: bug: snapshot/revertTo causes tests to pass unexpectedly
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (48d5d79 2022-09-01T00:10:22.933552Z)

### What command(s) is the bug in?

forge test

### Operating System

_No response_

### Describe the bug

If you run this test, it passes. If you comment out the `vm.revertTo` line it fails. Failure is expected in both cases since the assertion should fail. It seems like the cause is that reverting the state also reverts the `failed` flag: https://github.com/dapphub/ds-test/blob/9310e879db8ba3ea6d5c6489a579118fd264a3f5/src/test.sol#L47-L63
```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.15;
import "forge-std/Test.sol";

contract MyTest is Test {
  function test_snapshot() external {
    uint256 snapId = vm.snapshot();
    assertEq(uint256(0), uint256(1));
    vm.revertTo(snapId);
  }
}
```

### PR DESCRIPTION
**Pull request by mattsse**
## Motivation
Closes #3055

there were some changes in revm state recording, so we need to check the `JournaledState`'s present value of the slot that contains the `failed` variable when we revert a snapshot to record if there was a failure in a reverted snapshot

### CODE REVIEW

**Review by gakonst — APPROVED** [review]

**Comment by gakonst** [comment]
lgtm pending conflict fix

**Comment by onbjerg** [comment]
rebased, waiting for ci

**Pull request merged**
